### PR TITLE
chore: use testCtx instead of context.TODO() for scheduler integration

### PR DESCRIPTION
**Pull request by googs1025**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
- use testCtx instead of context.TODO() for scheduler integration
#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
None
```

### CODE REVIEW

**Review by sanposhiho — APPROVED** [review]
/lgtm 
/approve

**Comment by googs1025** [comment]
fix error print lint

File: test/integration/scheduler/scheduler_test.go [hunk 1]
```diff
@@ -561,7 +561,12 @@ func TestNodeEvents(t *testing.T) {
 	// 4. Remove the taint from node2; pod2 should now schedule on node2
 
 	testCtx := testutils.InitTestSchedulerWithNS(t, "node-events")
-	defer testCtx.ClientSet.CoreV1().Nodes().DeleteCollection(context.TODO(), metav1.DeleteOptions{}, metav1.ListOptions{})
+	defer func() {
+		err := testCtx.ClientSet.CoreV1().Nodes().DeleteCollection(testCtx.Ctx, metav1.DeleteOptions{}, metav1.ListOptions{})
+		if err != nil {
+			t.Fatalf("Failed to delete all node: %v", err)
+		}
+	}()
```
Comment (sanposhiho) [inline test/integration/scheduler/scheduler_test.go:569]: I guess we don't have to clean up nodes actually?
https://github.com/kubernetes/kubernetes/blob/712724634434cbf493ea48de6e711ab96e710cf3/test/integration/util/util.go#L548-L550

Comment (googs1025) [reply test/integration/scheduler/scheduler_test.go:569]: After reading the source code, it is indeed not needed. I will remove it

**Review by sanposhiho — APPROVED** [review]
/lgtm
/approve

**Pull request merged**